### ISSUE TITLE
docker-compose: logs polled on exited containers

### ISSUE THREAD

**Issue by mouchar** [tilt-dev/tilt#4904]
### Expected Behavior
When some docker-compose container exist correctly (exit code 0), Tilt should stop processing container logs. 

### Current Behavior
When d-c container exits, Tilt tries to follow its logs although the container is not running anymore.
`docker-compose logs --no-color -f containername` terminates immediately but Tilt tries to attach logs reader over and over.

### Steps to Reproduce
The minimal repoducer:
```
# Tiltfile:
docker_compose("./docker-compose.yml")

# ./docker-compose.yml
version: '3'
services:
  nginx:
    image: nginx
  one-off:
    image: alpine
    command: sh -c "date; echo Some command"
```
1. Run `tilt up`
2. Navigate to UI and check the `one-off` service
3. You'll see the very same line (with the same timestamp) repeating
4. check `pstree -a $(pgrep tilt)` (if available on your platform:
```
tilt up
  ├─docker-compose -f /tmp/test/docker-compose.yml logs --no-color -f nginx
  │   └─docker-compose -f /tmp/test/docker-compose.yml logs --no-color -f nginx
  │       └─2*[{docker-compose}]
  ├─docker-compose -f /tmp/test/docker-compose.yml logs --no-color -f one-off
  │   └─docker-compose -f /tmp/test/docker-compose.yml logs --no-color -f one-off
  └─18*[{tilt}]
```

CPU load is high because running `docker-compose logs` every second is a pretty expensive operation.

#### `tilt doctor` Output

```
$ tilt doctor
Tilt: v0.22.6, built 2021-08-27
System: linux-amd64
---
Docker
- Host: [default]
- Server Version: 20.10.7
- API Version: 1.41
- Builder: 2
---
Kubernetes
- Env: k3d
- Context: k3d-default
- Cluster Name: k3d-default
- Namespace: gooddata
- Container Runtime: containerd
- Version: v1.20.6+k3s1
- Cluster Local Registry: none
---
```
#### Docker-compose
```
$ docker-compose version
docker-compose version 1.25.4, build 8d51620a
docker-py version: 4.1.0
CPython version: 3.7.5
OpenSSL version: OpenSSL 1.1.0l  10 Sep 2019
```
#### About Your Use Case
Besides long-living services (db, app, web etc.) we also have several containers declared in our `docker-compose.yml`, that will just run one command and then exit. They have various purposes, like loading DB and app fixtures, checking that everything works, and so on).

We'd like to use Tilt to improve productivity for our developers while leveraging existing docker-compose manifests used for local development.

**Comment by milas** [comment]
Thanks for the detailed report! A fix for this along with several other improvements to Docker Compose log handling will be available in the next release.

**Comment by milas** [comment]
This is now fixed in [v0.22.8+](https://github.com/tilt-dev/tilt/releases/latest) - if you still see issues after upgrading, feel free to comment here or open a new issue!

As mentioned, there are several other improvements around Docker Compose log handling in this release as well - hopefully you see fewer "repeat" log lines and it's more robust at re-attaching if something goes wrong.
